src/ert/gui/model/snapshot.py

```diff
@@ -276,45 +283,55 @@ def _add_snapshot(self, snapshot: EnsembleSnapshot, iter_: str) -> None:
         self.rowsInserted.emit(parent, snapshot_tree.row(), snapshot_tree.row())
 
     @override
-    def columnCount(self, parent: QModelIndex | None = None) -> int:
+    def columnCount(
+        self, parent: QModelIndex | QPersistentModelIndex | None = None
+    ) -> int:
         if parent and isinstance(parent.internalPointer(), RealNode):
             return FM_STEP_COLUMN_SIZE
         return 1
 
-    def rowCount(self, parent: QModelIndex | None = None) -> int:
+    def rowCount(
+        self, parent: QModelIndex | QPersistentModelIndex | None = None
+    ) -> int:
         if parent is None:
             parent = QModelIndex()
         parent_item = self.root if not parent.isValid() else parent.internalPointer()
 
         if parent.column() > 0:
             return 0
 
-        return len(parent_item.children)
+        return len(parent_item.children)  # type: ignore
 
     @overload
-    def parent(self, child: QModelIndex) -> QModelIndex: ...
+    def parent(self) -> QObject: ...
     @overload
-    def parent(self) -> QObject | None: ...
+    def parent(self, child: QModelIndex | QPersistentModelIndex) -> QModelIndex: ...
     @override
-    def parent(self, child: QModelIndex | None = None) -> QObject | None:
+    def parent(
+        self, child: QModelIndex | QPersistentModelIndex | None = None
+    ) -> QObject | QModelIndex:
         if child is None or not child.isValid():
             return QModelIndex()
 
-        parent_item = child.internalPointer().parent
+        parent_item = child.internalPointer().parent  # type:ignore
         if parent_item == self.root:
             return QModelIndex()
 
         return self.createIndex(parent_item.row(), 0, parent_item)
 
     @override
-    def data(self, index: QModelIndex, role: int = Qt.ItemDataRole.DisplayRole) -> Any:
+    def data(
+        self,
+        index: QModelIndex | QPersistentModelIndex,
+        role: int = Qt.ItemDataRole.DisplayRole,
+    ) -> Any:
```

Comment:
Should we use Any?

Reply:
It is an override. It comes from QAbstractItemModel